ni_measurementlink_service/_datatypeinfo.py

```diff
@@ -43,14 +43,17 @@ def get_type_info(data_type: DataType) -> DataTypeInfo:
     DataType.Double: DataTypeInfo(type_pb2.Field.TYPE_DOUBLE, False),
     DataType.Boolean: DataTypeInfo(type_pb2.Field.TYPE_BOOL, False),
     DataType.String: DataTypeInfo(type_pb2.Field.TYPE_STRING, False),
-    DataType.Pin: DataTypeInfo(type_pb2.Field.TYPE_STRING, False, TypeSpecialization.Pin),
+    DataType.Pin: DataTypeInfo(type_pb2.Field.TYPE_STRING, False, TypeSpecialization.IOResource),
```

Comment:
I think changing the type specialization string from "pin" to "ioresource" breaks compatibility between the new ni-measurementlink-python and old versions of MeasurementLink.

Reply:
I spoke with @jasonmreding and @DelpireNI about this. We don't necessarily have clarity about which situations with new ni-measurementlink-python and old MeasurementLink we care about. Using a new DataType in a measurement (e.g. IOResource) wouldn't work with an old MeasurementLink, for example. But it does seem easier to run into trouble changing the behavior for an existing DataType, so I think it makes sense to keep this as "pin" as you suggest.

Reply:
I've replaced the Pin type specialization and maintain the use of it when using DataType.Pin to avoid the compatibility issue.

Reply:
Re: your deleted/edited comment: enums can have duplicates unless they're decorated with `@enum.unique`
https://docs.python.org/3/library/enum.html#enum.EnumType